src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/MuTect2.java

```diff
@@ -0,0 +1,1057 @@
+
+        readLikelihoods.changeReads(readRealignments);
+        for (final GATKRead rec : readRealignments.keySet()) {
+            logReadInfo(DEBUG_READ_NAME, rec, "Present after computing read likelihoods");
+        }
+
+        final HaplotypeCallerGenotypingEngine.CalledHaplotypes calledHaplotypes = genotypingEngine.callMutations(
+                readLikelihoods,
+                ARreads_origNormalMQ,
+                perSampleFilteredReadList,
+                assemblyResult.getFullReferenceWithPadding(),
+                assemblyResult.getPaddedReferenceLoc(),
+                regionForGenotyping.getSpan(),
+                featureContext,
+                header);
+
+        if ( MTAC.bamOutputPath != null ) {
+            final Set<Haplotype> calledHaplotypeSet = new HashSet<>(calledHaplotypes.getCalledHaplotypes());
+            if (MTAC.disableOptimizations)
+                calledHaplotypeSet.add(assemblyResult.getReferenceHaplotype());
+            haplotypeBAMWriter.writeReadsAlignedToHaplotypes(
+                    haplotypes,
+                    assemblyResult.getPaddedReferenceLoc(),
+                    haplotypes,
+                    calledHaplotypeSet,
+                    readLikelihoods);
+        }
+
+        if( MTAC.DEBUG ) { logger.info("----------------------------------------------------------------------------------"); }
+        return annotateVCs(calledHaplotypes, featureContext);
+    }
+
+    @Override
+    public void apply(final AssemblyRegion region, final ReferenceContext referenceContext, final FeatureContext featureContext ) {
+        final List<VariantContext> callsInRegion = callRegion(region, featureContext);
+
+        for ( final VariantContext call : callsInRegion ) {
+            // Only include calls that start within the current read shard (as opposed to the padded regions around it).
+            // This is critical to avoid duplicating events that span shard boundaries!
+            if ( getCurrentReadShardBounds().contains(new SimpleInterval(call.getContig(), call.getStart(), call.getStart())) ) {
+                vcfWriter.add(call);
+            }
+        }
+    }
+
+    private Set<String> calculateFilters(final FeatureContext featureContext, final VariantContext vc, final Map<String, Object> eventDistanceAttributes) {
+        final Set<String> filters = new HashSet<>();
+
+        final Integer eventCount = (Integer) eventDistanceAttributes.get(GATKVCFConstants.EVENT_COUNT_IN_HAPLOTYPE_KEY);
+        final Integer maxEventDistance = (Integer) eventDistanceAttributes.get(GATKVCFConstants.EVENT_DISTANCE_MAX_KEY);
+
+        //TODO: verify that vc.getStart() is the right second argument to use here
+        final Collection<VariantContext> panelOfNormalsVC = featureContext.getValues(MTAC.normalPanelFeatureInput, vc.getStart());
+        final VariantContext ponVc = panelOfNormalsVC.isEmpty()?null:panelOfNormalsVC.iterator().next();
+
+        if (ponVc != null) {
```

Comment:
I think the correct thing to do here is to filter only if the alleles match. Maybe the genotype is important too (e.g. if the variant in PON is hom var then don't filter).